### PR TITLE
Allow reading multiple blockette 55 entries (FAP) in dataless seed file

### LINKED ISSUE #3275: Xseed parser does not allow reading mutiple blockette 55 into ResponseListResponseStage object
### Avoid duplicates

- [X] I searched existing issues

### Bug Summary

Posted this in the obspy forum and thanks for the answer @megies 

I copied it here for further potential discussion and added test files. Have a fix that seems to work and will add a PR.

I stumbled over a (legacy) dataless seed response file where the responses are defined as frequency, amplitude, phase list (FAP) values.

The issue: Obspy read_inventory() reads only the first part of the FAP list. It turned out that multiple blockette 55 entries existed dividing the FAP response into different subsequent frequency ranges.

According to the SEED manual this is allowed: “…This blockette could exceed the maximum of 9999 characters. If so, continue with another blockette [55] on the next record. Set field 3 the same, but ignore fields 4 and 5 in subsequent blockettes.”

It seems in obspy.io.xseed.parser this is not taken into account, i.e. only the first blockette 55 is read for each channel in the dataless seed file.

Test files attached include a anonymized inventory of the station (because I am not sure if it can be shared). If you open the stationxml you see that the FAP list goes until 10 Hz. If you read the seed file with obspy it stops at 3 Hz.

[test_files.zip](https://github.com/obspy/obspy/files/10908028/test_files.zip)

(I create the xml file after I fixed the seed parser)

### Code to Reproduce

```python
inv=read_inventory('dummy_test_station.seed')
print(inv)
for stage in inv[0][0][0].response.response_stages:
    for f in stage.response_list_elements: print(f.frequency)
```

### Error Traceback

_No response_

### ObsPy Version?

1.3.0

### Operating System?

Linux

### Python Version?

3.10.4

### Installation Method?

conda

### PR DESCRIPTION
**Pull request by isbreen**
### What does this PR do?

Allows reading multiple blockette 55 entries (FAP respons list) in dataless seed file.

### Why was it initiated?  Any relevant Issues?

fixes #3275

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] While the PR is still work-in-progress, the `no_ci` label can be added to skip CI builds
- [/] If the PR is making changes to documentation, docs pages can be built automatically.
      Just add the `build_docs` tag to this PR.
      Docs will be served at [docs.obspy.org/pr/{branch_name}](https://docs.obspy.org/pr/) (do not use master branch).
      Please post a link to the relevant piece of documentation.
- [/] If all tests including network modules (e.g. `clients.fdsn`) should be tested for the PR,
      just add the `test_network` tag to this PR.
- [ ] All tests still pass.
- [/] Any new features or fixed regressions are covered via new tests.
- [x] Any new or changed features are fully documented.
- [/] Significant changes have been added to `CHANGELOG.txt` .
- [/] First time contributors have added your name to `CONTRIBUTORS.txt` .
- [/] If the changes affect any plotting functions you have checked that the plots
      from all the CI builds look correct. Add the "upload_plots" tag so that plotting 
      outputs are attached as artifacts. 
- [/] New modules, add the module to `CODEOWNERS` with your github handle
- [x] Add the `ready for review` tag when you are ready for the PR to be reviewed.

### CODE REVIEW

**Comment by megies** [comment]
@isbreen there are some file changes in taup data files in your PR. These need to be fixed.
This will also need a new test case. I'm not sure if this fits in maintenance branch, probably master is the better place for this, but that's not important right now, can be easily rebased later.

**Comment by isbreen** [comment]
> @isbreen there are some file changes in taup data files in your PR. These need to be fixed. This will also need a new test case. I'm not sure if this fits in maintenance branch, probably master is the better place for this, but that's not important right now, can be easily rebased later.

Not sure where these taup changes came from. I didn't touch them. So not sure how to fix this. Will try.

**Comment by megies** [comment]
Sometimes on other platforms weird stuff happens that adds like a line ending on the last line of files (there's differences between nix and windows there) and if you commit everything that just goes in there. That's why I always recommend `git add -p` which asks you for each part of the diff.

Don't worry about it now, I can fix that later. Can you change the test to use the smaller file I posted in the issue, though?

**Comment by isbreen** [comment]
> Sometimes on other platforms weird stuff happens that adds like a line ending on the last line of files (there's differences between nix and windows there) and if you commit everything that just goes in there. That's why I always recommend `git add -p` which asks you for each part of the diff.

 Ok, thanks for fixing this!

> Don't worry about it now, I can fix that later. Can you change the test to use the smaller file I posted in the issue, though?

I have exchanged the test data files with your files. Thanks for editing the dataless-seeds! That was out of my expertise.

File: obspy/io/xseed/tests/test_parser.py [hunk 1]
```diff
@@ -849,8 +849,8 @@ def test_parsing_resp_file_with_mutiple_blockette55(self, testdata):
         """
         Test case for issue #3275. Allow reading mutiple blockette 55.
         """
-        inv_xml = obspy.read_inventory('issue3275.xml')
-        inv_seed = obspy.read_inventory('issue3275.seed')
+        inv_xml = obspy.read_inventory(testdata['issue3275.xml'])
+        inv_seed = obspy.read_inventory(testdata['issue3275.seed'])
```
Comment (megies) [inline obspy/io/xseed/tests/test_parser.py]: @isbreen this is how to lookup test data within one module through the pytest fixture we provide, just fyi

**Review by megies — APPROVED** [review]
Did some cleanup and rebasing/squashing commits to clean up history. Should be good to go if CI goes green

**Comment by isbreen** [comment]
> Did some cleanup and rebasing/squashing commits to clean up history. Should be good to go if CI goes green

Thanks!

**Pull request merged**